osu.Game.Rulesets.Osu.Tests/TestSceneSliderSnaking.cs

```diff
@@ -0,0 +1,304 @@
+            AddAssert($"End vector {increased}", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                var last = body.CurrentCurve.Last();
+                return snaking ? last.X > vector.X && last.Y > vector.Y : last == vector;
+            });
+            addSeekStep(20100);
+            AddStep("Save start vector", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                vector = body.CurrentCurve.First();
+            });
+            addSeekStep(20200);
+            AddAssert("Start vector is same", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                var first = body.CurrentCurve.First();
+                return first == vector;
+            });
+            addSeekStep(23700);
+            AddStep("Save end vector", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                vector = body.CurrentCurve.Last();
+            });
+            addSeekStep(23800);
+            AddAssert("End vector is same", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                var last = body.CurrentCurve.Last();
+                return last == vector;
+            });
+            addSeekStep(27300);
+            AddStep("Save start vector", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                vector = body.CurrentCurve.First();
+            });
+            addSeekStep(27400);
+            AddAssert($"Start vector {increased}", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                var first = body.CurrentCurve.First();
+                return snaking ? first.X > vector.X && first.Y > vector.Y : first == vector;
+            });
+        }
+
+        private void setSnaking(bool value)
+        {
+            var text = value ? "Enable" : "Disable";
+            AddStep($"{text} snaking", () =>
+            {
+                snakingIn.Value = value;
+                snakingOut.Value = value;
+            });
+        }
+
+        private void addSeekStep(double time)
+        {
+            AddStep($"seek to {time}", () => track.Seek(time));
+
+            AddUntilStep("wait for seek to finish", () => Precision.AlmostEquals(time, Player.DrawableRuleset.FrameStableClock.CurrentTime, 100));
+        }
+
+        protected override IBeatmap CreateBeatmap(RulesetInfo ruleset) => new Beatmap
+        {
+            HitObjects = new List<HitObject>
+            {
+                new Slider
+                {
+                    StartTime = 3000,
+                    Position = new Vector2(100, 100),
+                    Path = new SliderPath(PathType.PerfectCurve, new[]
+                    {
+                        Vector2.Zero,
+                        new Vector2(300, 200)
+                    }),
```

Comment:
Seeing that we don't really care about position & path in these sliders I'd maybe extract out a helper method and take in `StartTime` and `RepeatCount` to make this a little leaner?

Reply:
Should the creation of beatmap also be automatic?